File: frontend/web/src/services/aiService.js
import request from '@/utils/request'

export const aiService = {
  // 生成回复建议
  generateReplySuggestions(postContent, existingComments = []) {
    return request.post('/ai/reply-suggestions', {
      postContent,
      existingComments
    })
  },

  // 总结帖子内容
  summarizePost(content) {
    return request.post('/ai/summarize', { content })
  },

  // 推荐标签
  recommendTags(title, content) {
    return request.post('/ai/recommend-tags', { title, content })
  },

  // 检测敏感内容
  checkSensitiveContent(content) {
    return request.post('/ai/check-sensitive', { content })
  },

  // 生成标题建议
  suggestTitle(content) {
    return request.post('/ai/suggest-title', { content })
  }
}

export default aiService

File: backend/src/main/java/com/zheminglt/service/AIService.java
package com.zheminglt.service;

import java.util.List;

public interface AIService {
    /**
     * 生成智能回复建议
     * @param postContent 帖子内容
     * @param existingComments 已有评论
     * @return 回复建议列表
     */
    List<String> generateReplySuggestions(String postContent, List<String> existingComments);

    /**
     * 智能总结帖子内容
     * @param content 帖子内容
     * @return 总结内容
     */
    String summarizePost(String content);

    /**
     * 自动推荐标签
     * @param title 帖子标题
     * @param content 帖子内容
     * @return 推荐标签列表
     */
    List<String> recommendTags(String title, String content);

    /**
     * 检测敏感内容
     * @param content 内容
     * @return 是否包含敏感内容
     */
    boolean containsSensitiveContent(String content);

    /**
     * 生成帖子标题建议
     * @param content 帖子内容
     * @return 标题建议
     */
    String suggestTitle(String content);
}
